File: Calculator/Calculator.cpp
#include <iostream>
#include <string>
#include <istream>
#include <iomanip>
#include <math.h>
using namespace std;

/* Initial User Calculation Input (returns full string of input) */
string inputString(){
	string phrase;
	cout << "Please enter a string for calculation. Include two numbers and a mathematical operator: ";
	cin >> phrase;
	return phrase;
}

/* Finds the operator being used in the string (returns string of a single operator) */
string findOperator(string phrase){
	string answer;
	string::size_type add = phrase.find("+", 0);
	string::size_type sub = phrase.find("-", 0);
	string::size_type mul = phrase.find("*", 0);
	string::size_type div = phrase.find("/", 0);

	if(add != std::string::npos) {
		answer = "+";
	}
	if(sub != std::string::npos) {
		answer = "-";
	}
	if(mul != std::string::npos) {
		answer = "*";
	}
	if(div != std::string::npos) {
		answer = "/";
	}
	return answer;
}

int main() {

		/* Input Calculation */
		string phrase = inputString();

		/* Find Operator */
		string op = findOperator(phrase);

		/* Find Numbers */
		std::string::size_type sz;   // alias of size_t

		//int num1 = std::stoi (phrase,&sz);

		//cout << num1 << endl;

		//double num1 = Double.parseDouble(phrase.substring(0,phrase.indexOf(operator)));
		//double num2 = Double.parseDouble(phrase.substring(phrase.indexOf(operator)+1));

		/* Calculate Answer */
		//string answer = findAnswer(num1,num2,op);
		//cout << answer;
	//}
	return(0);
}

File: Calculator/Calculator.java
import java.util.*;
public class Calculator{
	public static void main(String args[]){

		/* Input Calculation */
		String phrase = inputString();

		/* Find Operator */
		String operator = findOperator(phrase);

		/* Find Numbers */
		double num1 = Double.parseDouble(phrase.substring(0,phrase.indexOf(operator)));
		double num2 = Double.parseDouble(phrase.substring(phrase.indexOf(operator)+1));

		/* Calculate Answer */
		String answer = findAnswer(num1,num2,operator);
		System.out.println(answer);
	}
	
	/* Initial User Calculation Input (returns full string of input) */
	public static String inputString() {
		Scanner in = new Scanner(System.in);
		System.out.print("Please enter a string for calculation. Include two numbers and a mathematical operator: ");
		String answer = in.nextLine();
		return answer;
	}

	/* Finds the operator being used in the string (returns string of a single operator) */
	public static String findOperator(String phrase){
		String answer = "";
		int add_Index = phrase.indexOf("+");
		int sub_Index = phrase.indexOf("-");
		int mul_Index = phrase.indexOf("*");
		int div_Index = phrase.indexOf("/");

		if(add_Index != -1){
			answer = "+";
		} else if(sub_Index != -1) {
			answer = "-";
		} else if(mul_Index != -1) {
			answer = "*";
		} else if(div_Index != -1) {
			answer = "/";
		} else {
			answer = "error";
		}
		return answer;
	}

	/* Calculate input and return correct answer (returns String of equation input and answer) */
	public static String findAnswer(double num1, double num2, String op) {
		String answerString = "";
		double answer = 0.0;
		switch(op) {
			case "+":
				answer = num1 + num2;
				break;
			case "-":
				answer = num1 - num2;
				break;
			case "*":
				answer = num1 * num2;
				break;
			case "/":
				answer = num1 / num2;
				break;
			default:
				break;
		}
		answerString += num1 + " " + op + " " + num2 + " = " + answer;
		return answerString;
	}
}